general/development/policies/codingstyle/frankenstyle.md

```diff
@@ -0,0 +1,173 @@
+
+| Core subsystem | Frankenstyle component name | Location |
+| --- | --- | --- |
+| Access | core_access |  |
+| Administration | core_admin | /admin |
+| Antivirus | core_antivirus | /lib/antivirus |
+| Authentication | core_auth | /auth |
+| Conditional availability | core_availability | /availability |
+| Backup and restore | core_backup | /backup/util/ui |
+| Badges | core_badges | /badges |
+| Blocks | core_block | /blocks |
+| Blogging | core_blog | /blog |
+| Bulk users operations | core_bulkusers |  |
+| Caching | core_cache | /cache |
+| Calendar | core_calendar | /calendar |
+| Cohorts | core_cohort | /cohort |
+| Comment | core_comment | /comment |
+| Competency based education | core_competency | /competency |
+| Completion | core_completion | /completion |
+| Countries | core_countries |  |
+| Course | core_course | /course |
+| Currencies | core_currencies |  |
+| Database transfer | core_dbtransfer |  |
+| Debugging | core_debug |  |
+| Text editors | core_editor | /lib/editor |
+| Education fields | core_edufields |  |
+| Enrol | core_enrol | /enrol |
+| Error reporting | core_error |  |
+| Favourites | core_favourites | /favourites |
+| File picker | core_filepicker |  |
+| Files management | core_files | /files |
+| User filtering | core_filters |  |
+| Forms | core_form | /lib/form |
+| Grades | core_grades | /grade |
+| Advanced grading | core_grading | /grade/grading |
+| Groups | core_group | /group |
+| Help | core_help |  |
+| Hub | core_hub |  |
+| IMS CC | core_imscc |  |
+| Installer | core_install |  |
+| ISO 6392 | core_iso6392 |  |
+| Language pack configuration | core_langconfig |  |
+| License | core_license |  |
+| Maths library | core_mathslib |  |
+| Media | core_media |  |
+| Messaging | core_message | /message |
+| MIME types | core_mimetypes |
+| MNet | core_mnet | /mnet |
+| Dashboard | core_my | /my |
+| User notes | core_notes | /notes |
+| Page types | core_pagetype |  |
+| Pictures and icons | core_pix |  |
+| Plagiarism | core_plagiarism | /plagiarism |
+| Plugins management | core_plugin |  |
+| Portfolio | core_portfolio | /portfolio |
+| Privacy | core_privacy | /privacy |
+| Course publishing | core_publish | /course/publish |
+| Question bank engine | core_question | /question |
+| Ratings | core_rating | /rating |
+| Site registration | core_register | /admin/registration |
+| Repository | core_repository | /repository |
+| RSS | core_rss | /rss |
+| Roles | core_role | /admin/roles |
+| Global search | core_search | /search |
+| Tabular data display/download (deprecated) | core_table |  |
+| Tagging | core_tag | /tag |
+| Timezones | core_timezones |  |
+| User | core_user | /user |
+| User key | core_userkey |  |
+| Web service | core_webservice | /webservice |
+
+## Usage
+
+Frankenstyle component names are used in:
+
+### Function names
+
+All plugin functions must start with full frankenstyle prefix. For backwards compatibility modules may also use **modulename_** as prefix.
```

Comment:
We no longer recommend the use of global functions so I'd suggest adding a warning admonition:

```suggestion
All plugin functions must start with full frankenstyle prefix.

:::note Activity modules

For backwards compatibility modules may also use `modulename_` as prefix.

:::

:::warning

Something about global functions not being recommended. Please use an autoloaded class.

:::


```